src/Tests/Tests/Test_FastConv.cpp

```diff
@@ -18,22 +19,262 @@ namespace fastconv_test {
+            std::mt19937 generator(114514); // 114514 is seed
+            std::uniform_real_distribution<float> uniformRealDistribution(-1.F, 1.F);
+            for (int i = 0; i < IRLength; ++i) {
+                IR[i] = uniformRealDistribution(generator);
+            }
+
+            m_pCFastConv->init(IR, IRLength, IRLength, CFastConv::kTimeDomain);
         }
 
         virtual void TearDown()
         {
+            delete m_pCFastConv;
+            delete[] IR;
         }
 
-        CFastConv *m_pCFastConv = 0;
+        int IRLength = 0;
+        float* IR = nullptr;
+        CFastConv *m_pCFastConv = nullptr;
     };
 
-    TEST_F(FastConv, EmptyTest)
+    TEST_F(TimeDomainConv, identity)
+    {
+        // initialize input signal, which is $\delta[n-3]$
+        int shift = 3;
+        int signalLength = 10;
+        float* input = new float[signalLength]; // make the input signal 10 samples long
+        memset(input, 0, sizeof(float) * signalLength); // an impulse as input signal at sample index 3
+        input[shift] = 1.F;
+
+        // initialize output signal
+        float* output = new float[signalLength];
+        memset(output, 0, sizeof(float) * signalLength);
+
+        m_pCFastConv->process(output, input, signalLength);
+
+        // check if the output signal is exactly the same with the impulse response
+        for (int i = 0; i + shift < signalLength; ++i) {
+            EXPECT_NEAR(IR[i], output[i + shift], 1e-8);
+        }
+
+        // also check if the first three are ok
+        for (int i = 0; i < shift; ++i) {
+            EXPECT_EQ(output[i], 0);
+        }
+
+        delete[] input;
+        delete[] output;
+    }
+
+    TEST_F(TimeDomainConv, flushbuffer) {
+        // initialize input signal, which is $\delta[n-3]$
+        int shift = 3;
+        int signalLength = 10;
+        float* input = new float[signalLength]; // make the input signal 10 samples long
+        memset(input, 0, sizeof(float) * signalLength); // an impulse as input signal at sample index 3
+        input[shift] = 1.F;
+
+        // initialize output signal
+        float* output = new float[IRLength];
+        memset(output, 0, sizeof(float) * IRLength);
+
+        // process & discard result
+        m_pCFastConv->process(output, input, signalLength);
+        memset(output, 0, sizeof(float) * IRLength);
+
+        // flush
+        m_pCFastConv->flushBuffer(output);
+
+        // check if the tail is correct
+        for (int i = 0; i + signalLength - shift < IRLength; ++i) {
+            EXPECT_EQ(output[i], IR[i + signalLength - shift]);
+        }
+
+        delete[] input;
+        delete[] output;
+    }
+
+    TEST_F(TimeDomainConv, varyingBlockSize) {
+        // initialize input signal, which is $\delta[n-3]$
+        int shift = 3;
+        int signalLength = 10000;
+        float* input = new float[signalLength]; // make the input signal 10 samples long
```

Comment:
comment (maybe superfluous) doesn't match the code here

Reply:
Yeah, because this is duplicated from previous tests and I forgot to change comments😂😂. Thank you for pointing out this!

Reply:
Fixed in c57bd04eb77e91859acc32b437f5550b2f716eae.